infinispan-remote/src/test/java/org/hibernate/ogm/datastore/infinispanremote/utils/InfinispanRemoteTestHelper.java

```diff
@@ -178,23 +187,32 @@ public static void clearScriptStoredProcedures(SessionFactory sessionFactory) {
 		scriptCache.remove( "exceptionalProcedure" );
 	}
 
-	public static void deployJavaStoredProcedures() {
+	public static void deployJavaStoredProcedures() throws InterruptedException {
 		ShrinkWrap.create( JavaArchive.class, "simple-value-procedure.jar" )
 				.addClass( SimpleValueProcedure.class )
-				.addAsResource( asResource( SERVER_TASK_META_INF_RESOURCE_DIRECTORY + "/simple-value-procedure" ), SERVER_TASK_META_INF_TARGET_FILE )
+				.addAsResource( asResource( SERVER_TASK_SIMPLE_VALUE_PROCEDURE_META_INF ), SERVER_TASK_META_INF_TARGET_FILE )
 				.as( ZipExporter.class )
-				.exportTo( new File( INFINISPAN_DEPLOYMENTS_DIRECTORY + "/simple-value-procedure.jar" ), true );
+				.exportTo( new File( SERVER_TASK_SIMPLE_VALUE_PROCEDURE_JAR ), true );
 		ShrinkWrap.create( JavaArchive.class, "result-set-procedure.jar" )
 				.addClass( Car.class )
 				.addClass( ResultSetProcedure.class )
-				.addAsResource( asResource( SERVER_TASK_META_INF_RESOURCE_DIRECTORY + "/result-set-procedure" ), SERVER_TASK_META_INF_TARGET_FILE )
+				.addAsResource( asResource( SERVER_TASK_RESULT_SET_PROCEDURE_META_INF ), SERVER_TASK_META_INF_TARGET_FILE )
 				.as( ZipExporter.class )
-				.exportTo( new File( INFINISPAN_DEPLOYMENTS_DIRECTORY + "/result-set-procedure.jar" ), true );
-		ShrinkWrap.create( JavaArchive.class, INFINISPAN_DEPLOYMENTS_DIRECTORY + "exceptional-procedure.jar" )
+				.exportTo( new File( SERVER_TASK_RESULT_SET_PROCEDURE_JAR ), true );
+		ShrinkWrap.create( JavaArchive.class, "exceptional-procedure.jar" )
 				.addClass( ExceptionalProcedure.class )
-				.addAsResource( asResource( SERVER_TASK_META_INF_RESOURCE_DIRECTORY + "/exceptional-procedure" ), SERVER_TASK_META_INF_TARGET_FILE )
+				.addAsResource( asResource( SERVER_TASK_EXCEPTIONAL_PROCEDURE_META_INF ), SERVER_TASK_META_INF_TARGET_FILE )
 				.as( ZipExporter.class )
-				.exportTo( new File( INFINISPAN_DEPLOYMENTS_DIRECTORY + "/exceptional-procedure.jar" ), true );
+				.exportTo( new File( SERVER_TASK_EXCEPTIONAL_PROCEDURE_JAR ), true );
+		while ( !javaStoredProceduresDeployed() ) {
+			TimeUnit.SECONDS.sleep( 1 );
```

Comment:
What happens if the storedProcedures are never deployed? 
Have a lookt at HotRodServerRule to see an example of how we keep track of the number of attempts so that we don't end in an infinite loop.